backend/Buk.UniversalGames/Services/StatusService.cs

```diff
@@ -107,27 +142,30 @@ public async Task<List<TeamStatus>> BuildAndCacheLeagueRanking(int leagueId)
             var ticketTwist = (await GetGameRanking(GameType.TicketTwist, leagueId)).ToDictionary(x => x.TeamId);
             var minefield = (await GetGameRanking(GameType.MineField, leagueId)).ToDictionary(x => x.TeamId);
             var tableSurfing = (await GetGameRanking(GameType.TableSurfing, leagueId)).ToDictionary(x => x.TeamId);
+            
+            var sidequest = (await _cache.Get<List<TeamStatus>>("sidequest_ranking"))?.ToDictionary(x => x.TeamId) ?? new Dictionary<int, TeamStatus>();
 
             var leagueRanking = new List<TeamStatus>();
 
             foreach(var team in teams)
             {
-                var totalPoints = GetPointsForGame(nerveSpiral, team) 
-                                    + GetPointsForGame(monkeyBars, team)
-                                    + GetPointsForGame(ticketTwist, team)
-                                    + GetPointsForGame(minefield, team)
-                                    + GetPointsForGame(tableSurfing, team)
+                var totalPoints = GetPointsForSubRanking(nerveSpiral, team) 
```

Comment:
While this works fine AFAICS, it becomes kind of repetitive.

I'd replace all the copy-pasta with a few choice `Select` calls.

The games rankings could be an iterable over the `GameType` enum with a `.Select(async gameType => (await GetGameRanking(gameType, leagueId)).ToDictionary(x => x.TeamId))` which you can wrap in `Task.WhenAll` to await all asynchronous actions.

`leagueRanking` now is an assigned list followed by a `foreach` that ends with a `leagueRanking.Add` call. That's a perfect match for `var leagueRanking = teams.Select(team => …)

And if you have an iterable `gameRankings` list, you can make the points calculation a simple `gameRankings.Aggregate(_matchWinnerPoints * matchWinners.Count(x => x == team.TeamId), (totalPoints, gameRanking) => totalPoints + gameRanking.TryGetValue(team.TeamId, out var status) ? status.Points : 0);`

I'm not sure you can do `gameRanking.GetValueOrDefault(team.TeamId)?.Points ?? 0` but that would yield slightly clearer code then the ternary in my opinion. But that's very opinionated. I find null-propagating code easier to read than branching code. But that's mainly because my mind is a poor stack machine. I find it easier to read from left to right and have the type become nullable in a single-pass expression than to have to pop a context off my mental stack, and go mentally back to the ternary `?` once the true branch of a ternary ends, and the false branch begins.

> **Warning**
> Again, just off the top of my head, none of that code is guaranteed to compile.

Reply:
While the code is definately written with one-time use in mind, I'm also used to  favor what I consider easier to read code, which can mean some duplication or more lines of code within a function scope. I agree that a generic approach would work here as well, but don't see any definite improvements. I would probably consider to change in other codebases :).

Also keep in mind we're already adding the sidequest now, which is not a traditional game type and as such would need its own exception in this generic approach

Skipping the if/else structure whether tenary or traditional is not possible without helper methods, the method you're hoping for doesn't exist

Reply:
This can be marked as resolved. It was just a remark.

For my last remark, you don't need special helpers. I mentioned the wrong method name. The method I meant was `IEnumerable.FirstOrDefault(Func<bool> predicate)`. This takes advantage of the fact that `IDictionary<TKey, TValue>` is also an `IEnumerable<KeyValuePair<TKey, TValue>>`:

```csharp
gameRanking.FirstOrDefault(kvp =>  kvp.Key == team.TeamId)?.Value?.Points ?? 0
```